File: src/game/tools/rpc/output/wrap/GuildDemandWrap.cs
using System;
using Common.Net.Simple;

namespace Game 
{
	public class GuildDemandWrap : IWrapper
	{	
		public long MemberID; // 会员id
		public string Name; // 名字
		public int IconID; // 头像ID
		public long DemandTime; // 请求时间
		public long ExpireTime; // 过期时间
		public int HeroID; // 索要英雄
		public int MaxCount; // 最大数量
		public int CurCount; // 当前数量
		
		public void Decode(Packet pck)
		{	
			MemberID = pck.GetLong(); 
			Name = pck.GetString(); 
			IconID = pck.GetInt(); 
			DemandTime = pck.GetLong(); 
			ExpireTime = pck.GetLong(); 
			HeroID = pck.GetInt(); 
			MaxCount = pck.GetInt(); 
			CurCount = pck.GetInt(); 
		}
		
        public void Encode(Packet pck)
        {	
        	pck.PutLong(MemberID); 
        	pck.PutString(Name); 
        	pck.PutInt(IconID); 
        	pck.PutLong(DemandTime); 
        	pck.PutLong(ExpireTime); 
        	pck.PutInt(HeroID); 
        	pck.PutInt(MaxCount); 
        	pck.PutInt(CurCount); 
        }
	}
}

File: src/game/tools/rpc/output/wrap/GuildDemandWrap.go
package wraps

import (
	rpc "tipcat.com/common/rpc/simple"
)

type GuildDemandWrap struct {
	
	MemberID int64 // 会员id
	Name string // 名字
	IconID int32 // 头像ID
	DemandTime int64 // 请求时间
	ExpireTime int64 // 过期时间
	HeroID int32 // 索要英雄
	MaxCount int32 // 最大数量
	CurCount int32 // 当前数量
	rpc.Wrapper
}

func (w *GuildDemandWrap)Decode(pck *rpc.Packet) rpc.Wrapper {
	
	w.MemberID = pck.PopInt64(); 
	w.Name = pck.PopString(); 
	w.IconID = pck.PopInt32(); 
	w.DemandTime = pck.PopInt64(); 
	w.ExpireTime = pck.PopInt64(); 
	w.HeroID = pck.PopInt32(); 
	w.MaxCount = pck.PopInt32(); 
	w.CurCount = pck.PopInt32(); 
	return w
}

func (w *GuildDemandWrap)Encode(pck *rpc.Packet) {
	
	pck.PutInt64(w.MemberID) 
	pck.PutString(w.Name) 
	pck.PutInt32(w.IconID) 
	pck.PutInt64(w.DemandTime) 
	pck.PutInt64(w.ExpireTime) 
	pck.PutInt32(w.HeroID) 
	pck.PutInt32(w.MaxCount) 
	pck.PutInt32(w.CurCount) 
}

